File: Cap 09/9.3 - Estrutura de Registro de Matriz/9.3.1 a 9.3.4 juntos/regmatrix.java
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topico 9.3.1 ate 9.3.4
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ
 */

import java.util.Scanner;

class Main {
    static class BIMESTRE {
        double[] notas;
    }

    static class CAD_ALUNO {
        String nome;
        char turma;
        int sala;
        BIMESTRE notas;
    }

    public static void main(String[] args) {
        System.out.println("REGISTRO ESCOLAR - SIMPLES (1 ALUNO)");
        System.out.println();

        CAD_ALUNO aluno = new CAD_ALUNO();

        Scanner scanner = new Scanner(System.in);

        System.out.print("Entre o nome ......: ");
        aluno.nome = scanner.nextLine();

        System.out.print("Entre a turma .....: ");
        aluno.turma = scanner.nextLine().charAt(0);

        System.out.print("Entre a sala ......: ");
        aluno.sala = Integer.parseInt(scanner.nextLine());

        aluno.notas = new BIMESTRE();
        aluno.notas.notas = new double[4];
        for (int i = 0; i < 4; i++) {
            System.out.print("Entre a " + (i + 1) + "a nota ..: ");
            aluno.notas.notas[i] = Double.parseDouble(scanner.nextLine());
        }

        System.out.println();
        System.out.println("DADOS DO ALUNO");
        System.out.println();
        System.out.println("Nome ..............: " + aluno.nome);
        System.out.println("Turma .............: " + aluno.turma);
        System.out.println("Sala ..............: " + aluno.sala);
        for (int i = 0; i < 4; i++) {
            System.out.println("Nota " + (i + 1) + " ............: " + String.format("%.1f", aluno.notas.notas[i]));
        }
    }
}

File: Cap 09/9.3 - Estrutura de Registro de Matriz/9.3.1 a 9.3.4 juntos/regmatrix.kt
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topico 9.3.1 ate 9.3.4
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ
 * 
 */

import java.util.Scanner

typealias BIMESTRE = FloatArray

data class CAD_ALUNO(
    var NOME: String = "",
    var TURMA: Char = ' ',
    var SALA: Int = 0,
    var NOTAS: BIMESTRE = BIMESTRE(4)
)

fun main() {
  
    println("REGISTRO ESCOLAR - SIMPLES (1 ALUNO)\n")

    var ALUNO = CAD_ALUNO()

    var entrada = Scanner(System.`in`)

    print("Entre o nome ......: ")
    ALUNO.NOME = entrada.nextLine()

    print("Entre a turma .....: ")
    ALUNO.TURMA = entrada.nextLine()[0]

    print("Entre a sala ......: ")
    ALUNO.SALA = entrada.nextInt()

    for (I in 0 until 4) {
        print("Entre a ${I + 1}a. nota ..: ")
        ALUNO.NOTAS[I] = entrada.nextFloat()
    }

    println("\nDADOS DO ALUNO\n")
    println("Nome ..............: ${ALUNO.NOME}")
    println("Turma .............: ${ALUNO.TURMA}")
    println("Sala ..............: ${ALUNO.SALA}")
    for (I in 0 until 4) {
        println("Nota ${I + 1} ............: ${ALUNO.NOTAS[I]}")
    }
	
}
